corehq/apps/locations/permissions.py

```diff
@@ -319,6 +319,17 @@ def user_can_access_any_location_id(domain, user, location_ids):
             .exists())
 
 
+def user_can_access_other_user(domain, user, other_user):
+    if user.has_permission(domain, 'access_all_locations'):
+        return True
+
+    accessible_location_ids = (SQLLocation.objects
+        .accessible_to_user(domain, user)
+        .values_list('id', flat=True))
+
+    return other_user.sql_locations.filter(id__in=accessible_location_ids).count() > 0
```

Comment:
You can also chain these filters together to avoid making two db calls, and I believe `.exists()` is more performant than `.count() > 0`:
```python
return (other_user.get_sql_locations(domain)
        .accessible_to_user(domain, user)
        .exists())
```
There's also a function in this file called `user_can_access_any_location_id` which does very much the same thing (it's used only in `can_edit_form_location`).  It may be prudent to combine the two (this abstraction seems more appropriate, I can't think of another use-case for `access_any_location_id` beyond `access_other_user`).